### ISSUE TITLE
compilation error when rxcpp/rx.hpp is included after boost/asio.hpp on linux

### ISSUE THREAD

**Issue by hoffis** [Reactive-Extensions/RxCpp#165]
simple program below causes compile error on linux:

```cpp
#include <boost/asio.hpp>
#include <rxcpp/rx.hpp>

int main(int ,char**){
return 0;
}
```

The problem is that asio on linux include termios.h which in turn include bits/termios.h

bits/termios.h defindes B0 which conflicts with
rx-util.hpp 
all_true struct.

Solution would be change B0 to another name in rx-util.hpp

**Comment by kirkshoop** [comment]
I don't mind a pull request to change this, but #define B0 is hard to justify. Seems like termios needs an issue opened as well.  

On the side, I would really like to see code using asio and rxcpp together!

**Comment by hoffis** [comment]
Rxcpp and asio work together like a charm. Unfortunately We use them in commercial project that i'm not an owner so I can't show any code here. If I find some time, I will post some simple example of how those two can be used together.

**Comment by kirkshoop** [comment]
fixed in #168 

Thanks! 

Please let me know if you ever make that asio example.
